lib/streamlit/config.py

```diff
@@ -862,9 +862,17 @@ def _set_option(key: str, value: Any, where_defined: str) -> None:
     assert (
         _config_options is not None
     ), "_config_options should always be populated here."
-    assert key in _config_options, f'Key "{key}" is not defined.'
+    if key not in _config_options:
+        # Import logger locally to prevent circular references
+        from streamlit.logger import get_logger
 
-    _config_options[key].set_value(value, where_defined)
+        LOGGER = get_logger(__name__)
+
+        LOGGER.error(f' Key "{key}" is not defined.')
```

Comment:
nits:
* I think this should be a `LOGGER.warn` (or `LOGGER.warning`, I forget which of the two it is)
* There's a leading space in the log message

not a nit: The message should probably be changed to something more user-friendly now that it's a more expected thing to run into. Maybe something like

"{key} is not a valid config option. If you previously had this config option set, it may have been removed."

but it might be a good idea to check the wording with product

Reply:
Hmm I put a leading space because it looked ugly without the leading space. 
Without the leading space, it would look like `ERROR:streamlit.config:Key "not.defined" is not defined.`

Reply:
I feel like this shouldn't be the case since our log prefix should insert a leading space already. Running the server with the log level set to debug, you get output that looks like this:

```
022-02-03 09:48:38.684 DEBUG   streamlit.script_runner: Beginning script thread
2022-02-03 09:48:38.684 DEBUG   streamlit.script_runner: Running script RerunData(query_string='', widget_states=, page_name='', script_path='')
2022-02-03 09:48:38.684 DEBUG   streamlit.in_memory_file_manager: Disconnecting files for session with ID a76148db-5f66-4ece-8925-91655efe001c
...
```

which has a space between the calling function and the log message. None of these log calls have the leading space added manually